Plugins/Object/Object.cpp

```diff
@@ -594,4 +598,72 @@ ArgumentStack Object::SetTriggerGeometry(ArgumentStack&& args)
     return stack;
 }
 
+ArgumentStack Object::RemoveIconEffect(ArgumentStack&& args)
+{
+    ArgumentStack stack;
+
+    if (auto *pObject = object(args))
+    {
+        const auto nIcon = Services::Events::ExtractArgument<int32_t>(args);
+
+        for (int i = 0; i < pObject->m_appliedEffects.num; i++)
+        {
+            auto *eff = pObject->m_appliedEffects.element[i];
+
+            if (eff->m_sCustomTag == "NWNX_Object_IconEffect" && eff->m_nParamInteger[0] == nIcon)
+            {
+                pObject->RemoveEffect(eff);
+            }
+        }
+    }
+
+    return stack;
+}
+
+ArgumentStack Object::AddIconEffect(ArgumentStack&& args)
+{
+    ArgumentStack stack;
+
+    if (auto *pObject = object(args))
+    {
+        const auto nIcon = Services::Events::ExtractArgument<int32_t>(args);
```

Comment:
An `ASSERT_OR_THROW(nIcon > 0);` would be nice here

Reply:
Should I also add it to `RemoveIconEffect`?